src/runtime/node/timers/index.ts

```diff
@@ -0,0 +1,39 @@
+import { notImplemented, notImplementedAsync } from "../../_internal/utils";
+import noop from "../../mock/noop";
+import type timers from "node:timers";
+import promises from "./promises";
+
+export * as promises from "./promises";
+
+export const clearImmediate: typeof timers.clearImmediate =
+  globalThis.clearImmediate || noop;
+export const clearInterval: typeof timers.clearInterval =
+  globalThis.clearInterval || noop;
+export const clearTimeout: typeof timers.clearTimeout =
+  globalThis.clearTimeout || noop;
+
+export const setImmediate: typeof timers.setImmediate =
+  globalThis.setImmediate || notImplementedAsync("timers.setImmediate");
+export const setInterval: typeof timers.setInterval =
+  globalThis.setInterval || notImplementedAsync("timers.setInterval");
+export const setTimeout: typeof timers.setTimeout =
+  globalThis.setTimeout || notImplementedAsync("timers.setTimeout");
```

Comment:
I was thinking in js runtimes that timeouts don't make sense, perhaps we could immediately call the function as fallback? (if goal is compatibility, calling is better than throwing IMO)

Reply:
I understand the desire for getting the code working, but setTimeout is such a core api that turning it from an async to async call will likely cause more harm than good. I think it's highly unlikely that we'll ever see a runtime without `setInterval` and `setTimeout` in practice. So for these particular APIs we might as well drop the notImplementedAsync fallback to save a few bytes.

Reply:
Quick search: QuickJS and MuJS don't implement timers by default and such limited situations where runtimes don't have timers honestly make sense to me similar to work that sets `Date.now()` to `0` in ambient contexts.

PS: Let's consider when Node.js targeted libraries are used in build, they often pull in lots of more polyfills neither are bundle optimized, that few bytes is precision errors.

Reply:
What's the verdict here? `setTimeout || throw` or `setTimeout || call function as fallback`?

Reply:
Yep, and I think using second is totally safe for workerd at least..

Reply:
I'm not convinced about calling the fn as a fallback directly, but since won't have any impact on workerd, I'll defer to @pi0's decision here.

Reply:
Let's defer this until we need